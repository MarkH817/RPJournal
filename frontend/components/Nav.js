import React, { useState } from 'react'
import Link from 'next/link'
import styled from 'styled-components'

import { useUser } from './hooks/useUser'
import Logout from './Logout'

const StyledNav = styled.nav`
  font-size: 1.5em;

  a,
  button {
    background-color: transparent;
    border: none;
    display: block;
    font-family: inherit;
    font-size: inherit;
    line-height: inherit;
    padding: 0.25em 0.5em;
    text-decoration: none;

    &:hover,
    &:focus {
      text-decoration: underline;
    }
  }

  .menu-button {
    cursor: pointer;
    display: flex;
    align-items: center;
    text-align: left;
    width: 100%;
  }

  .menu {
    transition: all 0.3s ease-out;
    max-height: 0;
    overflow: hidden;

    &.active {
      max-height: 25em;
    }

    ul {
      border-top: 1px solid ${props => props.theme.black};
      list-style: none;
      margin: 0;
      padding: 0;
    }

    a,
    button {
      border-bottom: 1px solid ${props => props.theme.black};
    }

    button {
      color: ${props => props.theme.primaryColor};
      cursor: pointer;
      text-align: left;
      width: 100%;
    }
  }

  @media screen and (min-width: 55em) {
    .menu-button {
      display: none;
    }

    .menu {
      max-height: none;

      ul {
        border: 0;
        margin: 0 0 0 -0.25em;
      }

      li {
        display: inline-block;
        margin: 0 0.25em;
      }

      a,
      button {
        border: 0;
      }
    }
  }
`

export function Nav () {
  const [isActive, setActive] = useState(false)
  const { loading, error, data } = useUser()

  const toggleMenu = () => setActive(!isActive)
  const hideMenu = () => setActive(false)

  return (
    <StyledNav>
      <button className='menu-button' onClick={toggleMenu}>
        <span>Menu</span>
      </button>

      <div className={isActive ? 'menu active' : 'menu'} onClick={hideMenu}>
        <ul>
          <li>
            <Link href='/' prefetch>
              <a>Home</a>
            </Link>
          </li>

          <li>
            {loading || error || !data.me ? (
              <Link href='/login'>
                <a>Login</a>
              </Link>
            ) : (
              <Logout />
            )}
          </li>
        </ul>
      </div>
    </StyledNav>
  )
}

export default Nav
