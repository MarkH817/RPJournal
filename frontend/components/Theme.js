import React from 'react'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import 'modern-normalize'
import 'nprogress/nprogress.css'
import '@reach/skip-nav/styles.css'
import 'draft-js/dist/Draft.css'
import 'draftail/dist/draftail.css'
import 'typeface-open-sans'
import 'typeface-sarabun'

const theme = {
  white: 'rgb(255,255,255)',
  offWhite: 'rgb(225,225,225)',
  lightGray: 'rgb(175,175,175)',
  gray: 'rgb(125,125,125)',
  darkGray: 'rgb(75,75,75)',
  black: 'rgb(25,25,25)',
  primaryColor: '#ea45ba',
  accentColor: '#1d5e81',
  headerFont: `'Sarabun', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`,
  baseFont: `'Open Sans', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`
}

const GlobalStyles = createGlobalStyle`
  * {
    outline-width: 2px;
  }

  body {
    color: ${props => props.theme.black};
    font-family: ${props => props.theme.baseFont};
    font-size: 1em;
    padding: 0;
  }

  a {
    color: ${props => props.theme.primaryColor};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${props => props.theme.headerFont};
  }

  .detail-notes {
    line-height: 1.5;
  }
`

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <React.Fragment>
      <GlobalStyles />

      {children}
    </React.Fragment>
  </ThemeProvider>
)

export default Theme
