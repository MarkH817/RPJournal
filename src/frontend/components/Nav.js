import React from 'react'
import Link from 'next/link'
import styled from 'styled-components'

import User from './User'

const StyledNav = styled.nav`
  font-size: 1.5em;

  a,
  button {
    background-color: transparent;
    border: none;
    display: block;
    font-family: inherit;
    font-size: inherit;
    line-height: inherit;
    padding: 0.25em 0.5em;
    text-decoration: none;

    &:hover,
    &:focus {
      text-decoration: underline;
    }
  }

  .menu-button {
    cursor: pointer;
    display: flex;
    align-items: center;
    text-align: left;
    width: 100%;
  }

  .menu {
    transition: all 0.3s ease-out;
    max-height: 0;
    overflow: hidden;

    &.active {
      max-height: 25em;
    }

    ul {
      border-top: 1px solid ${props => props.theme.black};
      list-style: none;
      margin: 0;
      padding: 0;
    }

    a,
    button {
      border-bottom: 1px solid ${props => props.theme.black};
    }
  }

  @media screen and (min-width: 55em) {
    .menu-button {
      display: none;
    }

    .menu {
      max-height: none;

      ul {
        border: 0;
        margin: 0 0 0 -0.25em;
      }

      li {
        display: inline-block;
        margin: 0 0.25em;
      }

      a {
        border: 0;
      }
    }
  }
`

export class Nav extends React.PureComponent {
  state = { menuActive: false }

  toggleMenu = () =>
    this.setState(currentState => ({ menuActive: !currentState.menuActive }))

  hideMenu = () => this.setState({ menuActive: false })

  render () {
    const { menuActive } = this.state

    return (
      <StyledNav>
        <button className='menu-button' onClick={this.toggleMenu}>
          <span>Menu</span>
        </button>

        <div
          className={menuActive ? 'menu active' : 'menu'}
          onClick={this.hideMenu}
        >
          <ul>
            <li>
              <Link href='/' prefetch>
                <a>Home</a>
              </Link>
            </li>
            <li>
              <Link href='/new-adventure' prefetch>
                <a>New Adventure</a>
              </Link>
            </li>
            <User>
              {({ data: { me } }) =>
                !me ? (
                  <React.Fragment>
                    <li>
                      <Link href='/login' prefetch>
                        <a>Login</a>
                      </Link>
                    </li>
                  </React.Fragment>
                ) : (
                  <React.Fragment>
                    <li>
                      <button>Oh Hi, {me.name}!</button>
                    </li>
                  </React.Fragment>
                )
              }
            </User>
          </ul>
        </div>
      </StyledNav>
    )
  }
}

export default Nav
