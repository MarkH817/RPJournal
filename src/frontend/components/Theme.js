import React from 'react'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import 'modern-normalize'
import 'nprogress/nprogress.css'
import '@reach/skip-nav/styles.css'
import 'draft-js/dist/Draft.css'
import 'draftail/dist/draftail.css'
import 'typeface-open-sans'
import 'typeface-sarabun'

const theme = {
  white: '#ffffff',
  offWhite: '#dfdfdf',
  lightGray: '#c3c3c3',
  gray: '#767676',
  darkGray: '#525252',
  black: '#353535',
  primaryColor: '#ea45ba',
  accentColor: '',
  headerFont: `'Sarabun', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`,
  baseFont: `'Open Sans', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`
}

const GlobalStyles = createGlobalStyle`
  body {
    color: ${props => props.theme.black};
    font-family: ${props => props.theme.baseFont};
    font-size: 1em;
    padding: 0;
  }

  a {
    color: ${props => props.theme.primaryColor};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${props => props.theme.headerFont};
  }
`

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <React.Fragment>
      <GlobalStyles />

      {children}
    </React.Fragment>
  </ThemeProvider>
)

export default Theme
