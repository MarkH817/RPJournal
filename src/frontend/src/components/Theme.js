import React from 'react'
import '@reach/skip-nav/styles.css'
import { createGlobalStyle, ThemeProvider } from 'styled-components'

const theme = {
  black: '#353535',
  primaryColor: 'royalblue'
}

const GlobalStyles = createGlobalStyle`
  *,
  *:after,
  *:before {
    box-sizing: border-box;
  }

  body {
    color: ${props => props.theme.black};
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    font-size: 1em;
    margin: 0;
    padding: 0;
  }

  a {
    color: ${props => props.theme.primaryColor};
  }
`

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <React.Fragment>
      <GlobalStyles />

      {children}
    </React.Fragment>
  </ThemeProvider>
)

export default Theme
