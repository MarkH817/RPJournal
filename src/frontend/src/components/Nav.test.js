import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Nav from './Nav'
import Theme from './Theme'

describe('Nav', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Theme>
          <Nav />
        </Theme>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
