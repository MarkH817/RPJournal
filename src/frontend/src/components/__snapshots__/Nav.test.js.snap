// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
.c0 {
  font-size: 1.5em;
}

.c0 a,
.c0 button {
  background-color: transparent;
  border: none;
  display: block;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  padding: 0.25em 0.5em;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .menu-button {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: left;
  width: 100%;
}

.c0 .menu {
  -webkit-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
  max-height: 0;
  overflow: hidden;
}

.c0 .menu.active {
  max-height: 25em;
}

.c0 .menu ul {
  border-top: 1px solid #353535;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c0 .menu a {
  border-bottom: 1px solid #353535;
}

.c0 .menu a[aria-current] {
  border-bottom: 1px solid royalblue;
}

@media screen and (min-width:55em) {
  .c0 .menu-button {
    display: none;
  }

  .c0 .menu {
    max-height: none;
  }

  .c0 .menu ul {
    border: 0;
    margin: 0 0 0 -0.25em;
  }

  .c0 .menu li {
    display: inline-block;
    margin: 0 0.25em;
  }

  .c0 .menu a {
    border: 0;
  }
}

<nav
  className="c0"
>
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <span>
      Menu
    </span>
  </button>
  <div
    className="menu"
    onClick={[Function]}
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
